### PR TITLE
Fix problem with symlinked dlls and fix wmp11 package.

### PR DESCRIPTION
**Pull request by patlefort**
This PR fix some problems when a wine prefix contain symlinked dlls in its system32 or syswow64 folders.

* Add `w_try_cp_dll` function. This function will detect and remove symlinks before copying.
* Replace all `w_try cp` calls for `w_try_cp_dll` when they try to copy dlls into W_SYSTEM32_DLLS or W_SYSTEM64_DLLS.

wmp11 package:
* Remove dlls from both dll folders.
* Adding missing dlls.

Resident Evil on steam can now run. Tested with glorious eggroll proton. This also fix problem #1648.

There is probably a lot of other packages that will try to install stuff over symlinked dlls. That will cause one of 2 problems: Getting a permission denied if the target of the symlink is in a system folder without write access for the user. Or if the wine installation is writable by the user, its dlls will be overwritten, affecting all uses of that wine on all other prefixes, which is probably not what was intended.

### CODE REVIEW

**Comment by austin987** [comment]
```

+ grep -n -r [[:blank:]]$ src/winetricks
686:    
16656:    
src/winetricks contains trailing spaces, remove them. failed
```

Please fix that :)

**Comment by austin987** [comment]
It's still failing CI (now in shellcheck):
```

+ /usr/bin/shellcheck -s sh src/winetricks
======================== End shellcheck version info ===========================
Checking src/winetricks with shellcheck:

In src/winetricks line 685:
    [ -d "$2" ] && _W_destfile+="/$(basename "$1")"
                   ^---------^ SC2039: In POSIX sh, += is undefined.


In src/winetricks line 687:
    [ -h "$_W_destfile" ] && w_try rm -f "$_W_destfile"
          ^----------^ SC2250: Prefer putting braces around variable references even when not strictly required.
                                          ^----------^ SC2250: Prefer putting braces around variable references even when not strictly required.

Did you mean: 
    [ -h "${_W_destfile}" ] && w_try rm -f "${_W_destfile}"


In src/winetricks line 688:
    w_try cp -f "$1" "$_W_destfile"
                      ^----------^ SC2250: Prefer putting braces around variable references even when not strictly required.
```

To save time, try `./tests/shell-checks` locally before pushing :)

**Comment by patlefort** [comment]
Ran it through shellcheck this time, should be all good. I get some other warnings but unrelated to my changes.

File: src/winetricks [hunk 1]
```diff
@@ -676,6 +676,18 @@ w_try_cd()
     w_try cd "$@"
 }
 
+# Copy $1 into $2. If $2 is found to be a symbolic link, it will be removed first.
+# This solve a problem of dlls being symbolic links on some versions or variants of wine.
+# We want to replace the symbolic link and not copy into its target.
+w_try_cp_dll()
+{
+    _W_destfile="$2"
```
Comment (austin987) [inline src/winetricks:684]: Please give $1 a variable name and use it below as well (_W_srcfile/_W_srcdll).

File: src/winetricks [hunk 2]
```diff
@@ -676,6 +676,18 @@ w_try_cd()
...
+    [ -d "$2" ] && _W_destfile="${_W_destfile}/$(basename "$1")"
```
Comment (austin987) [inline src/winetricks:685]: `$2` should be `_W_destfile` (and `$1` the new variable).

File: src/winetricks [hunk 3]
```diff
@@ -16643,10 +16655,12 @@ load_wmp11()
...
 
     # remove builtin placeholders to allow update
-    w_try rm -f "${W_PROGRAMS_UNIX}/Windows Media Player/wmplayer.exe" "${W_SYSTEM64_DLLS}"/wmp.dll "${W_SYSTEM64_DLLS}"/wmvcore.dll
+    w_try rm -f "${W_PROGRAMS_UNIX}/Windows Media Player/wmplayer.exe" \
+        "${W_SYSTEM32_DLLS}"/wmp.dll "${W_SYSTEM32_DLLS}"/wmvcore.dll "${W_SYSTEM32_DLLS}"/mfplat.dll "${W_SYSTEM32_DLLS}"/wmasf.dll \
+        "${W_SYSTEM64_DLLS}"/wmp.dll "${W_SYSTEM64_DLLS}"/wmvcore.dll "${W_SYSTEM64_DLLS}"/mfplat.dll "${W_SYSTEM64_DLLS}"/wmasf.dll
 
     # need native overrides to allow update and later checks to succeed
-    w_override_dlls native l3codeca.acm wmp wmplayer.exe wmvcore wmpnssci
+    w_override_dlls native l3codeca.acm wmp wmplayer.exe wmvcore wmpnssci mfplat wmasf
```
Comment (austin987) [inline src/winetricks:16663]: Please keep these sorted.

**Review by austin987 — CHANGES_REQUESTED** [review]
Thanks, took a closer look now that CI passes ;).

Aside from above comments, some git housekeeping requests: 70c8d519cee789d37e8a96e7491db82977a1362b "Add w_try_cp_dll function" should be renamed/split (adding the function and using it together are fine, but the wmp11 overrides belong in a separate commit (e.g., "wmp11: add native mfplat/wmasf overrides").

The other commits IMO could then be squashed into the first commit.

**Comment by patlefort** [comment]
Should be cleaned up.

**Comment by roman-bronis** [comment]
It would be great if this got merged. It helped me to fix RE6 on Steam with latest Proton-GE (otherwise I was forced to use Proton 5.0). I believe this can help other games in the series as well.

**Comment by austin987** [comment]
Sorry for the delay; I wanted to make sure I ran the test suite before merging it ;).

**Pull request merged**